src/main/java/uk/gov/hmcts/ccd/data/IdListCacheKeyGenerator.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.ccd.data;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.tomcat.util.buf.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cache.interceptor.KeyGenerator;
+import org.springframework.stereotype.Component;
+
+@Component
+@Qualifier("idListCacheKeyGenerator")
+public class IdListCacheKeyGenerator implements KeyGenerator {
```

Comment:
Not for now. But I think a better approach in the future would be rather than caching the whole list of jurisdiction, to refactor the code to fetch 1 jurisdiction at the time so that we can cache the single Jurisdictions by jurisdiction id.
This will make better usage of the cache. Less cached data, because we can the single jurs rather that all the possible combinations of jur. And more cache hits.
But then will increase the number of requests to the backend in normal situations when cache is not used. So there are pros and cons. But worth pointing this out.

Reply:
Ok, removing this.